botorch/optim/parameter_constraints.py

```diff
@@ -312,13 +312,44 @@ def _make_linear_constraints(
     return constraints
 
 
+def _generate_unfixed_nonlin_constraints(
+    constraints: Optional[List[Callable]],
+    fixed_features: Dict[int, float],
+    dimension: int,
+) -> Optional[List[Callable]]:
```

Comment:
Two subjective suggestions to make this quicker to parse:
1) How about `_gen_nonlin_constraints_of_variable_features`? Could apply same convention to the linear case below. (cc @Balandat)
2) Could you add a doc-string here, something like "Given a dictionary of `fixed_features`, returns a list of non-linear constraints on the variable (non-fixed) features."

Reply:
Thank you! If you're up for it, you can do both in this commit. The only mentions of `_generate_unfixed_lin_constraints` are in:
- `generation/utils.py`,
- `optim/parameter_constraints.py`, and
- `test/optim/parameter_constraints.py`.
A quick search and replace in these files should do it.

Reply:
Hmm, but do you really prefer the new name? What do you do not like about `_generate_unfixed_lin_constraints`?

Reply:
But ofc, I can do it ;)

Reply:
`_generate_unfixed_(non)lin_constraints` makes it sound like the constraints are "unfixed", whereas it is the features that are. But let's keep it then, especially since the docstring makes it clear now. Thanks for adding it!